gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,715 @@
+  return McAlisterTripleSemigroup(G, OnPoints, X,
+                                  InducedSubdigraph(X, sub_ver));
+end);
+
+#############################################################################
+# Methods for McAlister triple semigroups
+#############################################################################
+InstallMethod(\=, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup, IsMcAlisterTripleSemigroup],
+function(x, y)
+  local A, G, X, Y, g;
+  G := McAlisterTripleSemigroupGroup(x);
+  X := McAlisterTripleSemigroupPartialOrder(x);
+  Y := McAlisterTripleSemigroupSemilattice(x);
+  if G = McAlisterTripleSemigroupGroup(y)
+      and X = McAlisterTripleSemigroupPartialOrder(y)
+      and Y = McAlisterTripleSemigroupSemilattice(y) then
+    if ForAll(G, g -> ForAll(DigraphVertices(X),
+         A -> McAlisterTripleSemigroupAction(x)(A, g) =
+              McAlisterTripleSemigroupAction(y)(A, g))) then
+      return true;
+    fi;
+  fi;
+  return false;
+end);
+
+InstallMethod(OneImmutable, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local Y;
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return MTE(S, DigraphSources(DigraphRemoveLoops(Y))[1], ());
+end);
+
+# (A, g) in S if and only if Ag^-1 is a vertex of the semilattice of S
+InstallMethod(AsList, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local out, g, A, V;
+    out := [];
+    V := DigraphVertexLabels(McAlisterTripleSemigroupSemilattice(S));
+    for g in McAlisterTripleSemigroupGroup(S) do
+      for A in V do
+        if (McAlisterTripleSemigroupAction(S)(A, Inverse(g)) in V) then
+          Add(out, MTE(S, A, g));
+        fi;
+      od;
+    od;
+    SetMcAlisterTripleSemigroupElmList(S, out);
+  return out;
+end);
+
+InstallMethod(String, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G, X, Y;
+  G := McAlisterTripleSemigroupGroup(S);
+  X := McAlisterTripleSemigroupPartialOrder(S);
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return Concatenation("McAlisterTripleSemigroup(", String(G), ", ",
+                       String(X), ", ", String(Y), ")");
+end);
+
+#TODO Linebreak hints
+
+InstallMethod(ViewString, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G;
+  G := McAlisterTripleSemigroupGroup(S);
+  return Concatenation("<McAlister triple semigroup over ", ViewString(G), ">");
+end);
+
+InstallMethod(IsomorphismSemigroups, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup, IsMcAlisterTripleSemigroup],
+function(S, T)
+  local iso_g, iso_x;
+
+  if not IsIsomorphicDigraph(McAlisterTripleSemigroupSemilattice(S),
+                             McAlisterTripleSemigroupSemilattice(T)) then
+    return fail;
+  fi;
+
+  iso_x := IsomorphismDigraphs(McAlisterTripleSemigroupPartialOrder(S),
+                               McAlisterTripleSemigroupPartialOrder(T));
```

Comment:
You are right, I need to check that. Should I do this by using `RestrictedMapping`... then is there something existing which can check if this is an isomorphism?